### ISSUE TITLE
Stuck (no response) periodically problem

### ISSUE THREAD

**Issue by canlang** [leonsoft-kras/transmisson-remote-gui#1021]
I tested all version (Linux, Mac, Windows)
Latest (maybe after 5.5) transmission remote version stuck (no response) periodically.
It seems to be a thread problem for the part that fetches the data.

Thanks,

**Comment by leonsoft-kras** [comment]
Many times have been answered. It's simple.  The TRGUI receives data from the transmission daemon using the RPC protocol. If the daemon is not loaded and is able to instantly return data, then the TRGUI will display them on the screen without delay.

Unload your demon. Limit its bandwidth (u/d). Disable upload / download. 
Will there be periodic problems?????????

P.S. TRGUI does not download or upload your torrents.  The program displays what the other program (transmission) is doing.

**Comment by leonsoft-kras** [comment]
There is no multithreading in TRGUI. So the author wrote the program. It is impossible to change programs - it must be completely rewritten from scratch.

I'm not an author. I'm just making small improvements.
Additionally, I modify the program to be compiled by a new compiler and for different operating systems. When I have time.

**Comment by canlang** [comment]
I understand this a remote program is just a front-end.

I looked for a commit in github with a clue of the phenomenon since 5.6v or later releases (and not OS depended). I'm not sure it's an inexperienced language base, but the **c76b5fd1aea707e6be9beea3827d2a3560af83a5** commit is doubtful.

**Comment by leonsoft-kras** [comment]
I can't help you. Too little information on your problem.

**Comment by canlang** [comment]
It' not matters. It's your open source.
Anyway, thanks for developing and have a nice day.

**Comment by leonsoft-kras** [comment]
Have a good day too.

For the future, report more information. 
If you can not repeat your situation, you can not find a problem in the code.

**Comment by antekgla** [comment]
Hi @canlang I´m responsable for mostly of the code of https://github.com/leonsoft-kras/transmisson-remote-gui/commit/c76b5fd1aea707e6be9beea3827d2a3560af83a5 commit.
I'm interest in your thoughts.
Please explain a little more about your problems.
Thanks!

**Comment by canlang** [comment]
Hi @antekgla 
Since I felt that the frequency of the freezing problem was similar to the period of fetching a torrent list, I estimated that a stuck would occur in an iterative loop fetching the torrent information. So I looked at commits after 5.5 and before 5.6 release. I thought that the **GetDisplayedTorrents**, **StatusBarSizes** part was created in the commit, and this part would cause a stuck phenomenon because it does not seem as a callback function.
Thank you!

**Comment by antekgla** [comment]
I would look into that code.

**Comment by antekgla** [comment]
Hi @canlang , the function **GetDisplayedTorrents** I made is inspired in GetSelectedTorrents what is in the original 5.0.1 code made by original author, only what mine only return the visible rows of torrent list.

Have you try to disable **Status Bar - Sizes** from the **View Menu**?
If you uncheck that, both functions **GetDisplayedTorrents** & **StatusBarSizes** are never executed so we can see if the problem is there.

My error was also not make that menu option persistent (saving in the transgui.ini) so you have to un-check this every time you run transgui.

**Comment by canlang** [comment]
Right. The size column seems to be a problem. Without that information, the phenomenon of the problem does not appear.

I have a question. In previous versions (5.5), the size column function was also present, why did you make it extra?

Thank you for reviewing.

**Comment by antekgla** [comment]
At wich size column you refer?
Size, Size to Download, Downloaded or Size Left?
The only colum not present in original 5.0.1 was Size Left and was added by @rzcat in https://github.com/leonsoft-kras/transmisson-remote-gui/commit/e4db54300400b20df3cdeb83cc532d36358453d3 March 26, 2016 and in release 5.3.0
I dont see problem with that code.
Can confirm what size colum is?

**Comment by canlang** [comment]
I checked **Size**, **Size to Download**, **Downloaded** and **Size Left** (4 item).
Only size column makes that problem.

**Comment by leonsoft-kras** [comment]
1. Can I see a short video of freezing the program with different column settings?

2. Can I see the screenshots of the program settings: 
    transmission option-> bandwidth, 
    application option-> general

**Comment by antekgla** [comment]
> Only the Size column makes that problem.

It not make any sense. That column remains unchanged from original 5.0.1 to today.
